enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,311 @@
+expected state, the services will be created or modified in order to restore expected state. Since the Node/ConfigMap
+controller could be reading from and modifying Windows service state, while these periodic state checks are made,
+appropriate thread safety procedures must be followed.
+
+### BYOH Node de-configuration: WMCO responsibilies
+
+When a user indicates to WMCO that it wishes to remove a BYOH node from the cluster, the following procedure will be
+kicked off.
+
+1) WMCO stops the WICD service on the instance.
+2) WMCO runs the cleanup command of WICD.
+3) WMCO removes the files it copied to the instance.
+
+Machine Nodes do not go through this process, as the Machine can be deleted and the Machine controller will drain
+and remove the Node.
+
+### BYOH Node de-configuration: WICD responsibilities
+
+When the WICD cleanup command is executed by WMCO, all services listed in the ConfigMap will be stopped and removed.
+It will then delete the instance's Node object.
+
+### Test Plan
+
+Most functionality described within this enhancement is already tested as part of the extensive end-to-end
+tests in the WMCO repo. As the different commands of WICD are developed, they can be integrated into WMCO, and
+tested through the existing tests. New tests will be added to test functionality that wasn't already present, such
+as validating the state of the services ConfigMap, and ensuring that Windows service state is reverted after it
+is tampered with.
+
+### Graduation Criteria
+
+This enhancement is targeted for OpenShift 4.11/ WMCO version 6.0.0.
+
+#### Dev Preview -> Tech Preview
+
+The features added here can be previewed in the Community 5.y.z release.
+
+#### Tech Preview -> GA
+
+As the functionality described in this enhancement will be integrated into WMCO, the normal release process
+of WMCO will be followed. That being a community operator release for early preview, followed by the official Red Hat
+operator release.
+
+#### Removing a deprecated feature
+
+N/A, as the changes made in the enhancement should be invisible to the user.
+
+### Upgrade / Downgrade Strategy
+
+When WMCO is upgraded, it will create the new Windows service ConfigMap, and then for each existing Windows Node:
+1) Stop the WICD service
+2) Run the WICD `cleanup` command with the `--upgrade` flag to stop and remove all services without removing the Node
+   object.
+2) Copy over all new and modified files in the payload, including WICD.
+3) Update the desired version annotation on the Node, to the current WMCO version.
+4) Start the WICD service.
+
+WICD, triggered by a mismatch between desired node version and current node version labels will correct the state
+by using the ConfigMap of the version it is upgrading to, in order to configure the instance normally. As part of this
+process, the Node version label is patched from the previous version to the desired version.
```

Comment:
You need to at least briefly call this out when you first introduce the desired version annotation.